build-tools/android-toolchain/android-toolchain.targets

```diff
@@ -117,4 +120,14 @@
         WorkingDirectory="$(MonoSourceFullPath)\sdks\builds"
     />
   </Target>
+
+  <Target Name="_CreateBuildConfig"
+      DependsOnTargets="GetNDKVersionInfo"
```

Comment:
There are no inputs and outputs except for the targets file itself, and it follows the pattern of `GetXAVersionInfo`

Reply:
The problem is that this target updates `XABuildConfig.cs`, which means we can have "cascading rebuilds": rebuild `android-toolchain`, `XABuildConfig.cs` is updated, then `Xamarin.Android.Build.Tasks.csproj` and `Xamarin.Android.Build.Tests.csproj` and all their dependencies need to be rebuilt.

This is a recipe for "Bad Things".

*At minimum*, we can have an `Inputs` of `..\scripts\XABuildConfig.cs.in` and `Outputs` of `..\..\bin\Build$(Configuration)\XABuildConfig.cs`.

That's not complete, because we need to invalidate when any of the properties changes, but it's a start.  We'd probably also want inputs of `build-tools\scripts\Ndk.targets` (for `$(AndroidNdkVersion)`), `$(AndroidNdkDirectory)\.stamp-ndk-$(AndroidNdkVersion)` (for the actual NDK install, which in turn controls `$(_NDKRevision)` and company, maybe?).

Another issue is that `<ReplaceFileContents/>` *always* updates the output file, even if it hasn't actually changed.

Thus, an "ideal" fix would be:

 1. Update `<ReplaceFileContents/>` so that the output file *isn't* touched if the contents haven't changed. See also `Files.CopyIfChanged()` (which might not be usable from the current `xa-prep-tasks.dll`, but that's fixable).
 2. Add Inputs & Outputs
 3. The Outputs should *not* be `XABuildConfig.cs`, but rather a "stamp" file.

For example:

```xml
<Target Name="_CreateBuildConfig"
    DependsOnTargets="GetNDKVersionInfo"
    AfterTargets="Build"
    Inputs="..\scripts\XABuildConfig.cs.in;..\scripts\Ndk.targets"
    Outputs="..\..\bin\Build$(Configuration)\.stamp-XABuildConfig.cs">
  <ReplaceFileContents
      SourceFile="..\scripts\XABuildConfig.cs.in"
      DestinationFile="..\..\bin\Build$(Configuration)\XABuildConfig.cs"
      Replacements="@NDK_MINIMUM_API_AVAILABLE@=$(_NDKMinimumApiAvailable);@NDK_RELEASE@=$(AndroidNdkVersion);@NDK_REVISION@=$(_NDKRevision);@NDK_VERSION_MAJOR@=$(_NDKVersionMajor);@NDK_VERSION_MINOR@=$(_NDKVersionMinor);@NDK_VERSION_MICRO@=$(_NDKVersionMicro);@NDK_ARMEABI_V7_API@=$(AndroidNdkApiLevel_ArmV7a);@NDK_ARM64_V8A_API@=$(AndroidNdkApiLevel_ArmV8a);@NDK_X86_API@=$(AndroidNdkApiLevel_X86);@NDK_X86_64_API@=$(AndroidNdkApiLevel_X86_64)"
  />
  <Touch Files="..\..\bin\Build$(Configuration)\.stamp-XABuildConfig.cs"  AlwaysCreate="True" />
</Target>
```

This would reduce/remove the risk of inadvertent rebuild cascades.